connectivity/netsocket/include/netsocket/NetworkInterface.h

```diff
@@ -101,6 +101,24 @@ class NetworkInterface: public DNS {
      */
     virtual const char *get_mac_address();
 
+    /** Set the MAC address to the interface.
+     *
+     *  Provided MAC address is set the the network interface. Address must be
+     *  set before calling the interface connect() method. EUI-48 MAC addresses
+     *  are used for Ethernet while Mesh interface is using 8-byte EUI-64 address.
+     *
+     *  All interfaces are not supporting MAC address set. A call to connect()
```

Comment:
can't the set_mac_address return an error if it's not supported?

Reply:
Not easily as the interface initialisation happens during the connect()-method call.

Reply:
If connect fails I think this makes quite inflexible interface as you can't recover from this and can't make a feature to read Mac from Factory tool, but for devices that has HW support for getting MAC it should override the factory tool configuration. So connect should not fail, but trace out an warning

Reply:
@mikter , good point!
Connect() failure is now ignored in the correction commit. Would you please review again?